anomaly-detection/src/main/java/io/pravega/anomalydetection/event/pipeline/EventStateMachineMapper.java

```diff
@@ -0,0 +1,49 @@
+/**
+ *
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries.
+ *
+ */
+package io.pravega.anomalydetection.event.pipeline;
+
+import io.pravega.anomalydetection.event.state.Event;
+import io.pravega.anomalydetection.event.state.EventStateMachine;
+import org.apache.flink.api.common.functions.FlatMapFunction;
+import org.apache.flink.streaming.api.checkpoint.Checkpointed;
+import org.apache.flink.util.Collector;
+
+import java.time.Instant;
+import java.util.HashMap;
+
+public class EventStateMachineMapper implements FlatMapFunction<Event, Event.Alert>, Checkpointed<HashMap<Integer,EventStateMachine.State>> {
```

Comment:
Yeah this code was ported from a relatively old codebase.   Not a big deal but it would be nice to demonstrate the best practice of using "managed state".